lightning/src/chain/chainmonitor.rs

```diff
@@ -153,19 +222,46 @@ where C::Target: chain::Filter,
 	pub fn get_claimable_balances(&self, ignored_channels: &[&ChannelDetails]) -> Vec<Balance> {
 		let mut ret = Vec::new();
 		let monitors = self.monitors.read().unwrap();
-		for (_, monitor) in monitors.iter().filter(|(funding_outpoint, _)| {
+		for (_, monitor_state) in monitors.iter().filter(|(funding_outpoint, _)| {
 			for chan in ignored_channels {
 				if chan.funding_txo.as_ref() == Some(funding_outpoint) {
 					return false;
 				}
 			}
 			true
 		}) {
-			ret.append(&mut monitor.get_claimable_balances());
+			ret.append(&mut monitor_state.monitor.get_claimable_balances());
 		}
 		ret
 	}
 
+	/// Gets the [`LockedChannelMonitor`] for a given funding outpoint, returning an `Err` if no
+	/// such [`ChannelMonitor`] is currently being monitored for.
+	///
+	/// Note that the result holds a mutex our monitor state, and should not be held indefinitely.
```

Comment:
Interesting -- is there ways in the bindings or in the native languages to `mem::drop` these objects? Might it be a limitation for future language bindings? :thinking: